Show diagnostics if using too old a version of Rust

Pretty new to Rust so forgive me if this is dumb in any way, but I was unable to build the demo:

## Expected Behavior
successful build

## Actual Behavior
unsuccessful build

$ cargo build --verbose
```
       Fresh version_check v0.9.2
       Fresh unicode-xid v0.2.1
       Fresh cc v1.0.66
       Fresh unicode-width v0.1.8
       Fresh glob v0.3.0
       Fresh termcolor v1.1.2
       Fresh lazy_static v1.4.0
       Fresh cfg-if v0.1.10
       Fresh cfg-if v1.0.0
       Fresh regex-syntax v0.6.21
       Fresh cxxbridge-flags v1.0.27
       Fresh vec_map v0.8.2
       Fresh ansi_term v0.11.0
       Fresh humantime v2.0.1
       Fresh strsim v0.8.0
       Fresh shlex v0.1.1
       Fresh unindent v0.1.7
       Fresh lazycell v1.3.0
       Fresh either v1.6.1
       Fresh peeking_take_while v0.1.2
       Fresh rustc-hash v1.1.0
       Fresh dunce v1.0.1
       Fresh thread_local v1.0.1
       Fresh libloading v0.6.6
       Fresh textwrap v0.11.0
       Fresh codespan-reporting v0.9.5
       Fresh indoc v1.0.3
       Fresh itertools v0.9.0
       Fresh proc-macro2 v1.0.24
       Fresh memchr v2.3.4
       Fresh quote v1.0.8
       Fresh aho-corasick v0.7.15
       Fresh regex v1.4.2
       Fresh log v0.4.11
       Fresh libc v0.2.81
       Fresh syn v1.0.58
       Fresh bitflags v1.2.1
       Fresh atty v0.2.14
       Fresh cxxbridge-macro v1.0.27
       Fresh which v3.1.1
       Fresh cxx-gen v0.7.27
       Fresh unzip-n v0.1.2
       Fresh proc-macro-error-attr v1.0.4
       Fresh clap v2.33.3
       Fresh env_logger v0.8.2
       Fresh nom v5.1.2
       Fresh link-cplusplus v1.0.4
       Fresh proc-macro-error v1.0.4
       Fresh cexpr v0.4.0
       Fresh clang-sys v1.0.3
       Fresh cxx v1.0.27
       Fresh autocxx-bindgen v0.56.1
       Fresh autocxx-engine v0.5.0
       Fresh autocxx-build v0.5.0
       Fresh autocxx-macro v0.5.0
       Fresh autocxx v0.5.0
   Compiling rust-itk-autocxx v0.1.0 (/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx)
     Running rustc --crate-name rust_itk_autocxx --edition=2018 src/main.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 -C metadata=87f4c50aab0614e8 -C extra-filename=-87f4c50aab0614e8 --out-dir /mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/deps -C incremental=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/incremental -L dependency=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/deps --extern autocxx=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/deps/libautocxx-e97985bbf2fadda9.rlib --extern cxx=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/deps/libcxx-441bc15e23333e54.rlib -L native=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/build/rust-itk-autocxx-3f4a93f2699b1a9a/out -l static=autocxx-demo -l c++ -L native=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/build/cxx-036b91c5b9563260/out -L native=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/build/link-cplusplus-b5d0ea38da856253/out
error: expected 'fn', found '{'
 --> src/main.rs:4:1
  |
4 | / include_cpp! {
5 | |     #include "input.h"
6 | |     safety!(unsafe_ffi)
7 | |     generate!("DoMath")
8 | | }
  | |_^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: use of undeclared type or module `ffi`
  --> src/main.rs:11:59
   |
11 |     println!("Hello, world! - C++ math should say 12={}", ffi::DoMath(4));
   |                                                           ^^^ use of undeclared type or module 'ffi'

error: aborting due to 2 previous errors

For more information about this error, try 'rustc --explain E0433'.
error: could not compile 'rust-itk-autocxx'.

Caused by:
  process didn't exit successfully: 'rustc --crate-name rust_itk_autocxx --edition=2018 src/main.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 -C metadata=87f4c50aab0614e8 -C extra-filename=-87f4c50aab0614e8 --out-dir /mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/deps -C incremental=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/incremental -L dependency=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/deps --ex tern autocxx=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/deps/libautocxx-e97985bbf2fadda9.rlib --extern cxx=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/deps/libcxx-441bc15e23333e54.rlib -L native=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/build/rust-itk-autocxx-3f4a93f2699b1a9a/out -l static=autocxx-demo -l c++ -L native=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/build/cxx-036b91c5b9563260/out -L native=/mnt/yard/william.nitsch/projects/image-analysis-common/rust/rust-itk-autocxx/target/debug/build/link-cplusplus-b5d0ea38da856253/out' (exit code: 1)

```
## Steps to Reproduce the Problem

1. cargo build

## Specifications

- Version: 0.5.0
- Platform: Amazon Linux 5.4.74-36.135.amzn2.x86_64

--- a/engine/Cargo.toml
+++ b/engine/Cargo.toml
@@ -40,6 +40,7 @@ unzip-n = "0.1.2"
 cxx-gen = "0.7.27"
 cxx = "1.0.27"
 autocxx-parser = { path="../parser" }
+version_check = "0.9"
 
 [dependencies.syn]
 version = "1.0.39"

--- a/engine/src/builder.rs
+++ b/engine/src/builder.rs
@@ -110,6 +110,7 @@ where
     I: IntoIterator<Item = T>,
     T: AsRef<OsStr>,
 {
+    rust_version_check();
     let gen_location_strategy = match custom_gendir {
         None => FileLocationStrategy::new(),
         Some(custom_dir) => FileLocationStrategy::Custom(custom_dir),
@@ -212,4 +213,10 @@ fn write_rs_to_file(
     content: TokenStream,
 ) -> Result<PathBuf, BuilderError> {
     write_to_file(dir, filename, content.to_string().as_bytes())
+}
+
+fn rust_version_check() {
+    if !version_check::is_min_version("1.48.0").unwrap_or(false) {
+        panic!("Rust 1.48 or later is required.")
+    }
 }

